### ISSUE TITLE
[Networklogger] Incorrect Typescript signature for NetworkLogger.setNetworkDataObfuscationHandler(...)

### ISSUE THREAD

**Issue by tomwanzek** [Instabug/Instabug-React-Native#580]
### Steps to Reproduce the Problem

The typescript signature provided for `NetworkLogger.setNetworkDataObfuscationHandler(...)` here:

https://github.com/Instabug/Instabug-React-Native/blob/2c895b2b0ad2b971856f09df0137435d194d0498/index.d.ts#L142

seems inaccurate.

### Expected Behavior

As per the documentation here https://docs.instabug.com/docs/react-native-logging#obfuscating-data and actual behaviour, it seems an async mapping function is expected.

Something of the sort

```typescript
(networkData: InstabugNetworkData) => Promise<InstabugNetworkData>
```

While I indicated some appropriately defined `InstabugNetworkData` for the network data that are mapped, it is by no means explicit in any of the published documentation/TypesScript definitions, what the data structure is actually defined as.

We had to "piece" a skeleton definition together based on actual data flowing through, so we can work in TypeScript with the SDK.

### Actual Behavior

Currently the typescript definition specifies a `handler` argument of type `() => void` which one needs to hack around to make the method usable in typescript.

There is no TypeScript definition for the network data  object.

### Instabug integration code

n/a

### SDK Version

10.0.0

### React Native, iOS and Android Versions

all

Thanks in advance for fixing/enhancing the definitions to have a more robust development experience for such a sensitive part of the Instabug SDK implementation. 🙇 😄

**Comment by AliAbdelfattah** [comment]
@tomwanzek thank you for reporting the issue. We're working on a fix.

**Comment by AliAbdelfattah** [comment]
@tomwanzek we've released a fix in v10.4. It shouldn't throw an error now.
